pkg/agent/controller/networkpolicy/networkpolicy_controller_test.go

```diff
@@ -436,3 +456,102 @@ func TestAddNetworkPolicyWithMultipleRules(t *testing.T) {
+	clientset.AddWatchReactor("addressgroups", k8stesting.DefaultWatchReactor(addressGroupWatcher, nil))
+	clientset.AddWatchReactor("appliedtogroups", k8stesting.DefaultWatchReactor(appliedToGroupWatcher, nil))
+	clientset.AddWatchReactor("networkpolicies", k8stesting.DefaultWatchReactor(networkPolicyWatcher, nil))
+
+	protocolTCP := v1beta1.ProtocolTCP
+	port := intstr.FromInt(80)
+	services := []v1beta1.Service{{Protocol: &protocolTCP, Port: &port}}
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	go controller.Run(stopCh)
+
+	// Test adding policy1 with a single rule
+	policy1 := newNetworkPolicy("policy1", []string{"addressGroup1"}, []string{}, []string{"appliedToGroup1"}, services)
+	networkPolicyWatcher.Add(policy1)
+	networkPolicyWatcher.Action(watch.Bookmark, nil)
+	select {
+	case ruleID := <-reconciler.updated:
+		t.Fatalf("Expected no update, got %v", ruleID)
+	case <-time.After(time.Millisecond * 100):
+	}
+	checkNetworkPolicyMetrics(t, controller)
+
+	// Test adding policy2 with multiple rules
+	policy2 := newNetworkPolicyWithMultipleRules("policy2", []string{"addressGroup2"}, []string{"addressGroup2"}, []string{"appliedToGroup2"}, services)
+	networkPolicyWatcher.Add(policy2)
+	select {
+	case ruleID := <-reconciler.updated:
+		t.Fatalf("Expected no update, got %v", ruleID)
+	case <-time.After(time.Millisecond * 100):
+	}
+	checkNetworkPolicyMetrics(t, controller)
+
+	// Test deleting policy1
+	networkPolicyWatcher.Delete(newNetworkPolicy("policy1", []string{}, []string{}, []string{}, nil))
+	select {
+	case ruleID := <-reconciler.updated:
+		t.Fatalf("Expected no update, got %v", ruleID)
+	case <-time.After(time.Millisecond * 100):
+	}
+	checkNetworkPolicyMetrics(t, controller)
+
+	// Test deleting policy2
+	networkPolicyWatcher.Delete(newNetworkPolicy("policy2", []string{}, []string{}, []string{}, nil))
+	select {
+	case ruleID := <-reconciler.updated:
+		t.Fatalf("Expected no update, got %v", ruleID)
+	case <-time.After(time.Millisecond * 100):
+	}
+	checkNetworkPolicyMetrics(t, controller)
+}
+
+func checkNetworkPolicyMetrics(t *testing.T, controller *Controller) {
+	expectedEgressNetworkPolicyRuleCount := `
+    # HELP antrea_agent_egress_networkpolicy_rule_count [STABLE] Number of egress networkpolicy rules on local node which are managed by the Antrea Agent.
+    # TYPE antrea_agent_egress_networkpolicy_rule_count gauge
+	`
+
+	expectedIngressNetworkPolicyRuleCount := `
+    # HELP antrea_agent_ingress_networkpolicy_rule_count [STABLE] Number of ingress networkpolicy rules on local node which are managed by the Antrea Agent.
+    # TYPE antrea_agent_ingress_networkpolicy_rule_count gauge
+	`
+
+	expectedNetworkPolicyCount := `
+    # HELP antrea_agent_networkpolicy_count [STABLE] Number of networkpolicies on local node which are managed by the Antrea Agent.
+    # TYPE antrea_agent_networkpolicy_count gauge
+	`
+
+	ingressRuleCount := 0
+	egressRuleCount := 0
+
+	networkpolicies := controller.GetNetworkPolicies("")
```

Comment:
add a comment ("get policies in all namespaces"), I had to look up in the code what `""` means in this context